### ISSUE TITLE
Using PyTorch Dataset, DataLoader instead of TimeSeriesDataSet

### ISSUE THREAD

**Issue by eTuDpy** [jdb78/pytorch-forecasting#901]
I am trying to implement a use case in which `time_varying_known_reals` have a different sampling rate as the `target`. Therefore, the implemented `TimeSeriesDataSet` class does not suit me. I am trying to solve this by using a custom `Dataset` Class. Everything is implemented and operates on the classical PyTorch `Dataset` Class, however, my resulting `DataLoader` is still different from the one you get with `TimeSeriesDataSet .to_dataloader()`. What else do I need to adjust? For instance, how do I include the automatic training / testing feature? I have included a training / testing split, but my `__getitem__` only works for training data. 

This is my custom `DataSet` Class:
(It is still a work in progress, some adjustable variables are still a little hard coded, however, the core runs as it should)

    class MaxRealDataSet(Dataset):
    def __init__(self, df_in, df_out, numerical_cols, target_col, categorical_cols=[], freq=1, seq_length=10,
                 prediction_window=6):
        # self.df_in = df_in
        # self.df_out = df_out
        self.target_col = target_col
        self.prediction_window = prediction_window
        self.numerical_cols = numerical_cols
        self.categorical_cols = categorical_cols
        self.preprocessor = None
        self.seq_length = int((seq_length * 60) / freq)
        self.step_size = 100
        
        ## Parameter
        self.train_size = 0.8

        ## Reindex
        df_in.loc[:, 'index'] = np.arange(0, df_in.shape[0] * df_in['index'][1],
                                          df_in['index'][1], dtype='int64')
        df_out.loc[:, 'index'] = np.arange(0, df_out.shape[0] * df_out['index'][1],
                                           df_out['index'][1], dtype='int64')

        ## Preprocessing
        self.X_train_df, self.X_test_df, self.y_train_df, self.y_test_df = self.preprocess_data(df_in, df_out)

        # self.frame_series(self.X_train_df, self.y_train_df)
        # self.get_item_test(idx=3)

    def preprocess_data(self, df_in, df_out):
        ## Normalization
        if self.categorical_cols == []:
            self.preprocessor_in = ColumnTransformer([("scaler_out", StandardScaler(), self.numerical_cols)],
                                                     remainder="drop")
        else:
            self.preprocessor_in = ColumnTransformer([("scaler_in", StandardScaler(), self.numerical_cols),
                                                      ("encoder", OneHotEncoder(handle_unknown='ignore'),
                                                       self.numerical_cols)],
                                                     remainder="drop")

        self.preprocessor_out = ColumnTransformer([("scaler_out", StandardScaler(), self.target_col)],
                                                  remainder="drop")

        ## Split
        X_train_df = df_in[:int(self.train_size * df_in.shape[0])]
        X_test_df = df_in[int(self.train_size * df_in.shape[0]):]
        y_train_df = df_out[:int(self.train_size * df_out.shape[0])]
        y_test_df = df_out[int(self.train_size * df_out.shape[0]):]

        X_train_df.loc[:, self.numerical_cols] = self.preprocessor_in.fit_transform(X_train_df)
        y_train_df.loc[:, self.target_col] = self.preprocessor_out.fit_transform(y_train_df)
        X_test_df.loc[:, self.numerical_cols] = self.preprocessor_in.transform(X_test_df)
        y_test_df.loc[:, self.target_col] = self.preprocessor_out.transform(y_test_df)

        return X_train_df, X_test_df, y_train_df, y_test_df

    def get_item_test(self, idx):
        ## Windowing
        X_idx = torch.tensor(self.X_train_df.loc[int(self.step_size * idx):
                                                 int(self.step_size * idx + self.seq_length - 1),
                             self.numerical_cols].values)
        y_idx = torch.tensor(self.y_train_df.loc[idx + int(self.seq_length / 100):
                                                 idx + int(self.seq_length / 100) + self.prediction_window - 1,
                             self.target_col].values)
        return X_idx, y_idx

    def __getitem__(self, idx):
        ## Windowing
        X_idx = torch.tensor(self.X_train_df.loc[int(self.step_size * idx):
                                                 int(self.step_size * idx + self.seq_length - 1),
                             self.numerical_cols].values)
        y_idx = torch.tensor(self.y_train_df.loc[idx + int(self.seq_length / 100):
                                                 idx + int(self.seq_length / 100) + self.prediction_window - 1,
                             self.target_col].values)
        return X_idx, y_idx

    def __len__(self):
        return self.y_train_df.shape[0] - int(self.seq_length / 100) - self.prediction_window
